File: app/models/comment.py
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base
from app.models.post import Post
from app.models.user import User


class Comment(Base):
    __tablename__ = "comments"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    info: Mapped[str] = mapped_column(Text)
    post_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("posts.id", ondelete="CASCADE"), index=True
    )
    owner_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("users.id", ondelete="CASCADE"), index=True
    )
    blocked_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), index=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )

    post: Mapped[Post] = relationship()
    owner: Mapped[User] = relationship()

File: web/src/types/comment.ts
export type CommentDto = {
  id: number;
  info: string;
  post_id: number;
  owner_id: number;
  blocked_at: string | null;
  created_at: string;
};

export type Comment = {
  id: number;
  info: string;
  ownerId: number;
  createdAt: string;
  isMine: boolean;
  isBlocked: boolean;
};

export type CommentWrite = {
  info: string;
};
